src/rbyte/io/table/mcap/ros2_reader.py

```diff
@@ -0,0 +1,117 @@
+import json
+from collections.abc import Hashable, Mapping
+from functools import cached_property
+from mmap import ACCESS_READ, mmap
+from operator import attrgetter
+from os import PathLike
+from pathlib import Path
+from typing import Any, Literal, cast, override
+
+import more_itertools as mit
+import polars as pl
+from mcap.reader import SeekingReader
+from mcap_ros2.decoder import DecoderFactory
+from optree import tree_flatten_with_path
+from pydantic import ConfigDict, ImportString
+from structlog import get_logger
+from xxhash import xxh3_64_intdigest as digest
+
+from rbyte.config.base import BaseModel, HydraConfig
+from rbyte.io.table.base import TableReaderBase
+
+logger = get_logger(__name__)
+
+
+PolarsDataType = pl.DataType | pl.DataTypeClass
+
+
+Leaf = HydraConfig[PolarsDataType] | ImportString[PolarsDataType] | None
+
+
+class Config(BaseModel):
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    # HACK: limit to three levels deep  # noqa: FIX004
+    fields: Mapping[str, Mapping[str, Leaf | Mapping[str, Leaf | Mapping[str, Leaf]]]]
+
+    validate_crcs: bool = False
+
+
+class McapRos2TableReader(TableReaderBase, Hashable):
+    FRAME_INDEX_COLUMN_NAME: Literal["frame_idx"] = "frame_idx"
```

Comment:
This can't be tested in this PR correct ?

Reply:
i did test this, need an appropriate mcap file tho (https://jkk-research.github.io/dataset/jkk_dataset_02/)